fix: Fix chat functionality on non-Wynncraft servers [skip ci]

P0keDev opened the pull request:
The event in `ChatListenerMixin` was set to always fire instead of only firing while online Wynncraft, which stopped chat messages from appearing if the player was on a different server. The change in `ScreenMixin` was done to prevent chat tab buttons from appearing outside of Wynncraft when the Minecraft window was resized while chat was open.

P0keDev committed:
fix ScreenInit and ClientsideMessage events to only fire while on wynncraft

kristofbolyai reviewed (approved):
I assume you checked that all other mixins use the helper. Can you just glance through the usages of the events to make sure we don't depend on them for determining we are on Wynn. Afaik not, and I'll take your word for it.

kristofbolyai commented:
And importantly, thanks!

P0keDev committed:
update LightTextureMixin to use MixinHelper

P0keDev committed:
revert ScreenMixin change

P0keDev commented:
Found an issue caused in `WynncraftButtonFeature` by the changes to `ScreenMixin`, so I reverted that for now. I also double checked and found a second mixin that wasn't using the helper function, so I fixed that one too.
